qiskit_nature/problems/second_quantization/lattice/lattice/hyper_cubic_lattice.py

```diff
@@ -0,0 +1,251 @@
+                )
+
+        super().__init__(graph)
+
+        # default position for one and two-dimensional cases.
+        if self.dim == 1:
+            if self.boundary_conditions[0] == "open":
+                self.pos = {i: [i, 0] for i in range(self.size[0])}
+            elif self.boundary_conditions[0] == "periodic":
+                theta = 2 * pi / self.size[0]
+                self.pos = {i: [np.cos(i * theta), np.sin(i * theta)] for i in range(self.size[0])}
+        elif self.dim == 2:
+            self.pos = {}
+            for index in range(np.prod(self.size)):
+                # maps an index to two-dimensional coordinate
+                # the positions are shifted so that the edges between boundaries can be seen
+                # for the periodic cases.
+                x = index % self.size[0]
+                y = index // self.size[0]
+                if self.boundary_conditions[1] == "open":
+                    return_x = x
+                elif self.boundary_conditions[1] == "periodic":
+                    return_x = x + 0.2 * np.sin(pi * y / (self.size[1] - 1))
+                if self.boundary_conditions[0] == "open":
+                    return_y = y
+                elif self.boundary_conditions[0] == "periodic":
+                    return_y = y + 0.2 * np.sin(pi * x / (self.size[0] - 1))
+                self.pos[index] = [return_x, return_y]
+
+    # pylint: disable=missing-param-doc
+    def draw_without_boundary(
+        self,
+        self_loop: bool = False,
+        pos: Optional[dict] = None,
+        ax: Optional[Axes] = None,
+        arrows: bool = True,
+        arrowstyle: Optional[str] = None,
+        arrow_size: int = 10,
+        with_labels: bool = False,
+        node_list: Optional[list] = None,
+        edge_list: Optional[list] = None,
+        node_size: Union[int, list] = 300,
+        node_color: Union[
+            str,
+            Tuple[float, float, float],
+            Tuple[float, float, float],
+            List[Union[str, Tuple[float, float, float], Tuple[float, float, float, float]]],
+        ] = "#1f78b4",
+        node_shape: str = "o",
+        alpha: Optional[float] = None,
+        cmap: Optional[Colormap] = None,
+        vmin: Optional[float] = None,
+        vmax: Optional[float] = None,
+        linewidths: Union[float, Sequence] = 1.0,
+        width: Union[float, Sequence] = 1.0,
+        edge_color: Union[str, Sequence] = "k",
+        edge_cmap: Optional[Colormap] = None,
+        edge_vmin: Optional[float] = None,
+        edge_vmax: Optional[float] = None,
+        style: str = "solid",
+        labels: Optional[Callable] = None,
+        edge_labels: Optional[Callable] = None,
+        font_size: int = 12,
+        font_color: str = "k",
+        font_weight: str = "normal",
+        font_family: str = "sans-serif",
+        label: Optional[str] = None,
+        connectionstyle: str = "arc3",
```

Comment:
I still think we should try to eliminate the duplication of all of these arguments...